fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryLockTest.java

```diff
@@ -87,4 +89,29 @@ void testFileLock(boolean useAgile) throws IOException {
             lock2.close();
         }
     }
+
+    @ParameterizedTest
+    @CsvSource({"true,true", "true,false", "false,true", "false,false"})
+    void testFileLockClose(boolean useAgile, boolean abortAgilityContext) throws IOException {
+        try (FDBRecordContext context = fdb.openContext()) {
+            AgilityContext agilityContext =
+                    useAgile ?
+                    AgilityContext.agile(context, 1000, 100_0000) :
+                    AgilityContext.nonAgile(context);
+
+            FDBDirectory directory = new FDBDirectory(subspace, null, null, null, true, agilityContext);
```

Comment:
You should also close the directory after closing the lock.

Reply:
✅